app/code/Magento/Downloadable/Helper/Download.php

```diff
@@ -252,10 +253,21 @@ public function setResource($resourceFile, $linkType = self::LINK_TYPE_FILE)
                 );
             }
         }
-
+        
         $this->_resourceFile = $resourceFile;
+        
+        /**
+        * check header for urls
+        */
+        if ($this->_linkType === self::LINK_TYPE_URL) {
```

Comment:
@shikhamis11 please pay attention that `_linkType` property is set later in this method. I think it'll be better to compare to `$linkType`

Reply:
Yes you are right I just updated with out noticing that let me correct this